pkg/policy/groups/helpers.go

```diff
@@ -123,24 +172,74 @@ func updateDerivativeStatus(cnp *cilium_v2.CiliumNetworkPolicy, derivativeName s
 		status.OK = true
 	}
 
+	if clusterScoped {
+		return updateDerivativeCNPStatus(cnp, status, derivativeName)
+	}
+
+	return updateDerivativeCCNPStatus(cnp, status, derivativeName)
```

Comment:
The logic here seems flipped, `clusterScoped == true` should pair with `CCNP`, not the other way around?

Reply:
Yeah, good catch.